logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -222,10 +222,13 @@ public IRubyObject ruby_to_hash_with_metadata(ThreadContext context) {
             return Rubyfier.deep(context.runtime, data);
         }
 
-        @JRubyMethod(name = "to_java")
-        public IRubyObject ruby_to_java(ThreadContext context)
-        {
-            return JavaUtil.convertJavaToUsableRubyObject(context.runtime, this.event);
+        @Override // this will make to_java ruby return a Java event (proxy)
+        @SuppressWarnings("unchecked")
+        public <T> T toJava(final Class<T> target) {
```

Comment:
acts as the previous `to_java` on the Ruby side by hooking up the `toJava` (interface) contract.

Reply:
I'm pretty confused about naming here. I would have expected `to_java` to return a usable _java_ object, but under the hood the original converted to a usable _ruby_ object with `JavaUtil.convertJavaToUsableRubyObject`?

Reply:
default `to_java` relies on the [toJava method](https://github.com/jruby/jruby/blob/9.2.17.0/core/src/main/java/org/jruby/java/addons/KernelJavaAddons.java#L18), there's actually 2 ways `toJava` is used:
- to handle auto-converting of any `IRubyObject` with Java scripting, whenever there's a `callJava(String)` seen by the runtime an object (whether it's wrapped Java or a Ruby one e.g. `RubyString` instance) gets passed by doing a `callJava( obj.toJava(String.class) )`
- if there's an explicit Ruby land `obj.to_java` it also uses `toJava` and wraps the returned value into a Java proxy if necessary (same as `JavaUtil.convertJavaToUsableRubyObject` previously here)